NitroxClient/GameLogic/Bases/BuildUtils.cs

```diff
@@ -0,0 +1,318 @@
+            if (mapRoomFunctionality)
+            {
+                // As MapRooms can be built as the first piece of a base, we need to make sure that they receive a new id if they're not in a base
+                if (constructableBase.GetComponentInParent<Base>(true))
+                {
+                    NitroxEntity.SetNewId(mapRoomFunctionality.gameObject, id);
+                }
+                else
+                {
+                    NitroxEntity.SetNewId(mapRoomFunctionality.gameObject, id.Increment());
+                }
+                moduleObject = mapRoomFunctionality.gameObject;
+                return true;
+            }
+            Log.Error($"Couldn't find MapRoomFunctionality of built MapRoom (cell: {face.Value.cell})");
+            moduleObject = null;
+            return false;
+        }
+
+        IBaseModule module = baseGhost.targetBase.GetModule(face.Value);
+        if (module != null)
+        {
+            // If the WaterPark is higher than one, it means that the newly built WaterPark will be merged with one that already has a NitroxEntity
+            if (module is WaterPark waterPark && waterPark.height > 1)
+            {
+                Log.Debug($"Found WaterPark higher than 1 [{waterPark.height}], not transferring NitroxEntity to it");
+                // as the WaterPark is necessarily merged, we won't need to do anything about it
+                moduleObject = null;
+                return false;
+            }
+
+            Log.Debug($"Successfully transferred NitroxEntity to {module} [{id}]");
+            moduleObject = (module as Component).gameObject;
+            NitroxEntity.SetNewId(moduleObject, id);
+            return true;
+        }
+        // When a WaterPark is merged with another one, we won't find its module but we don't care about that
+        else if (!isWaterPark)
+        {
+            Log.Error("Couldn't find the module's GameObject of built interior piece when transfering its NitroxEntity to the module.");
+        }
+
+        moduleObject = null;
+        return false;
+    }
+
+    /// <remarks>
+    ///     The criteria to make sure that a ConstructableBase is one of a BaseDeconstructable is if it has a moduleFace
+    ///     because this field is only filled for the base deconstruction (<see cref="BaseDeconstructable.Deconstruct"/>, <seealso cref="ConstructableBase.LinkModule(Base.Face?)"/>).
+    /// </remarks>
+    public static bool IsBaseDeconstructable(ConstructableBase constructableBase)
+    {
+        return constructableBase.moduleFace.HasValue;
+    }
+
+    /// <remarks>
+    /// A BaseDeconstructable's ghost component is a simple BaseGhost so we need to identify it by the parent ConstructableBase instead.
+    /// </remarks>
+    /// <param name="faceAlreadyLinked">Whether <see cref="ConstructableBase.moduleFace"/> was already set or not</param>
```

Comment:
Fixed the other way around (you or someone else asked me to go with "faceAlreadyLinked" instead of "faceNotLinked" but I completely forgot to change the related references to it which caused a regression (will be fixed with next commits)